test/gtest/kernel_tuning_net.cpp

```diff
@@ -10,71 +10,83 @@ struct KernelTuningNetTestCase : AIModelTestCase
     std::string arch;
 };
 
-std::vector<KernelTuningNetTestCase> GetConvAsm1x1UTestCases()
+std::vector<KernelTuningNetTestCase> GetConvAsm1x1UTestCases_FP32()
 {
     return {{{{1, 512, 192, 288, {56, 56}, {1, 1}, {0, 0}, {1, 1}, {1, 1}},
               miopen::conv::Direction::BackwardData,
               miopenFloat,
               miopenTensorNCHW},
              "1,16,1,64,2,2,1,4",
-             "gfx908"},
-            {{{1, 256, 2048, 512, {7, 7}, {1, 1}, {0, 0}, {1, 1}, {1, 1}},
+             "gfx908"}};
+}
+std::vector<KernelTuningNetTestCase> GetConvAsm1x1UTestCases_FP16()
+{
+    return {{{{1, 256, 2048, 512, {7, 7}, {1, 1}, {0, 0}, {1, 1}, {1, 1}},
               miopen::conv::Direction::Forward,
               miopenHalf,
               miopenTensorNCHW},
              "2,8,4,16,1,4,1,4",
              "gfx908"}};
 }
 
-std::vector<KernelTuningNetTestCase> GetConvHipIgemmGroupFwdXdlopsTestCases()
+std::vector<KernelTuningNetTestCase> GetConvHipIgemmGroupFwdXdlopsTestCases_FP16()
 {
-    return {
-        {{{1, 128, 64, 128, {209, 209}, {3, 3}, {0, 0}, {2, 2}, {1, 1}},
-          miopen::conv::Direction::Forward,
-          miopenFloat,
-          miopenTensorNHWC},
-         "DeviceGroupedConvFwdMultipleABD_Xdl_CShuffle<256, 128, 128, 16, Default, 32, 32, 2, 2, "
-         "4, 4, 4, 1, 1, 1>",
-         "gfx90a"},
-        {{{16, 256, 2016, 192, {7, 7}, {1, 1}, {0, 0}, {1, 1}, {1, 1}},
-          miopen::conv::Direction::Forward,
-          miopenHalf,
-          miopenTensorNHWC},
-         "DeviceGroupedConvFwdMultipleABD_Xdl_CShuffle<64, 64, 64, 32, Filter1x1Stride1Pad0, 32, "
-         "32, 2, 2, 1, "
-         "1, 1, 1, 1>",
-         "gfx942"},
-    };
+    return {{{{1, 128, 64, 128, {209, 209}, {3, 3}, {0, 0}, {2, 2}, {1, 1}},
+              miopen::conv::Direction::Forward,
+              miopenFloat,
+              miopenTensorNHWC},
+             "DeviceGroupedConvFwdMultipleABD_Xdl_CShuffle<256, 128, 128, 16, Default, 32, 32, 2, "
+             "2, 4, 4, 4, 1, 1, 1>",
+             "gfx90a"}};
+}
+
+std::vector<KernelTuningNetTestCase> GetConvHipIgemmGroupFwdXdlopsTestCases_FP32()
+{
+    return {{{{16, 256, 2016, 192, {7, 7}, {1, 1}, {0, 0}, {1, 1}, {1, 1}},
+              miopen::conv::Direction::Forward,
+              miopenHalf,
+              miopenTensorNHWC},
+             "DeviceGroupedConvFwdMultipleABD_Xdl_CShuffle<64, 64, 64, 32, Filter1x1Stride1Pad0, "
+             "32, 32, 2, 2, 1, 1, 1, 1, 1>",
+             "gfx942"}};
 }
```

Comment:
Are the data types here flipped?

Looks like FP16 has miopenFloat, and FP32 has miopenHalf.

Reply:
:facepalm: